app/src/main/java/com/google/droidjump/GameView.java

```diff
@@ -16,16 +16,184 @@
 
 package com.google.droidjump;
 
+import static androidx.navigation.Navigation.findNavController;
+import android.annotation.SuppressLint;
 import android.content.Context;
+import android.graphics.Canvas;
+import android.graphics.Color;
+import android.graphics.Paint;
+import android.os.Bundle;
+import android.util.Log;
+import android.view.MotionEvent;
+import android.view.SurfaceHolder;
 import android.view.SurfaceView;
 
 public class GameView extends SurfaceView implements Runnable {
 
-    public GameView(Context context) {
+    private int currentLevel;
+    private MainActivity activity;
+    private boolean isPlaying;
+    private int screenX;
+    private int screenY;
+    private int screenMargin;
+    private SurfaceHolder surfaceHolder;
+    private Droid droid;
+    private int timePoint;
+    private Thread thread;
+    private int levelTimePoints;
+    private int levelSpeed;
+    private Bundle arguments;
+    private Paint levelPaint;
+
+    public GameView(Context context, int screenX, int screenY, Bundle arguments, boolean isPlaying) {
         super(context);
+        receiveLevelDetails();
+        activity = (MainActivity) context;
+        timePoint = 0;
+        surfaceHolder = getHolder();
+        currentLevel = getCurrentLevel(arguments);
+        this.arguments = arguments;
+        this.screenX = screenX;
+        this.screenY = screenY;
+        this.isPlaying = isPlaying;
+        levelPaint = createLevelPaint();
+        screenMargin = (int) getResources().getDimension(R.dimen.fab_margin);
+
+        // Create droid
+        droid = new Droid(screenMargin, screenY - screenMargin, levelSpeed, getResources());
+        droid.setDroidJumpHeight(getResources());
+    }
+
+    public Paint createLevelPaint() {
+        Paint paint = new Paint();
+        paint.setAntiAlias(true);
+        paint.setColor(Color.BLACK);
+        paint.setTextSize(getResources().getDimension(R.dimen.header_text_size));
+        return paint;
+    }
+
+    public int getCurrentLevel(Bundle arguments) {
+        return arguments.getInt("level", activity.getCurrentLevel());
```

Comment:
nit: have a string variable to which contains "level" instead of a hardcoded string.

Instead of just having level, consider having the path to this file +level. eg. "java.com.google.droidjump.GameView.Level" -- so that we minimise the risk of using the same string twice.